### ISSUE TITLE
Rename tf.nn.batch_normalization

### ISSUE THREAD

**Issue by Smokrow** [tensorflow/tensorflow#26284]
**System information**
- TensorFlow version: 1.13
- Doc Link: https://www.tensorflow.org/api_docs/python/tf/nn/batch_normalization

**Describe the documentation issue**
tf.nn.batch_normalization can actually be used to implement layer_normalization or group_normalization as well. (https://github.com/tensorflow/addons/pull/14)  Therefore the name of the function is quite confusing since the "batch normalization" is more or less a special case of this function (when you feed the right mean and variance tensor). 

I think something like "normalization()" could work as well and would be less confusing.

**Comment by Ayush517** [comment]
@dynamicwebpaige Should I rename?
